src/node_sqlite.cc

```diff
@@ -678,6 +698,110 @@ void DatabaseSync::CustomFunction(const FunctionCallbackInfo<Value>& args) {
   CHECK_ERROR_OR_THROW(env->isolate(), db->connection_, r, SQLITE_OK, void());
 }
 
+// database.aggregate(name, { start, step, inverse, result, directOnly,
+// deterministic, varargs })
+void DatabaseSync::AggregateFunction(const FunctionCallbackInfo<Value>& args) {
+  DatabaseSync* db;
+  ASSIGN_OR_RETURN_UNWRAP(&db, args.This());
+  Environment* env = Environment::GetCurrent(args);
+  THROW_AND_RETURN_ON_BAD_STATE(env, !db->IsOpen(), "database is not open");
+
+  Utf8Value name(env->isolate(), args[0].As<String>());
+  Local<Object> options = args[1].As<Object>();
+  Local<Value> step_v;
+  if (!options->Get(env->context(), env->step_string()).ToLocal(&step_func)) {
+    // TODO: throw an error; step is required
```

Comment:
If Get returns nothing and this is false, then v8 should already be scheduling an error to be thrown. There shouldn't be a need to throw another error.

Reply:
Oh, great! Thanks for letting me know!